subgraphs/_reference_/src/prices/routers/CurveRouter.ts

```diff
@@ -1,120 +1,280 @@
-    );
-
-    const basePrice = getBasePrice(curveLpTokenAddress, curveRegistryContract);
-    if (basePrice.reverted) continue;
-
-    return CustomPriceType.initialize(
-      price.times(basePrice.usdPrice),
-      basePrice.decimals
-    );
-  }
-
-  return new CustomPriceType();
-}
-
-export function getPoolFromLpToken(
-  lpAddress: Address,
-  curveRegistry: CurveRegistryContract
-): Address {
-  const poolAddress = utils.readValue<Address>(
-    curveRegistry.try_get_pool_from_lp_token(lpAddress),
-    constants.NULL.TYPE_ADDRESS
-  );
-
-  return poolAddress;
-}
-
-export function getBasePrice(
-  curveLpTokenAddress: Address,
-  curveRegistry: CurveRegistryContract
-): CustomPriceType {
-  const poolAddress = getPoolFromLpToken(curveLpTokenAddress, curveRegistry);
-
-  if (poolAddress.equals(constants.NULL.TYPE_ADDRESS)) {
-    return new CustomPriceType();
-  }
-
-  const underlyingCoinAddress = getUnderlyingCoinFromPool(
-    poolAddress,
-    curveRegistry
-  );
-
-  const basePrice = getPriceUsdcRecommended(underlyingCoinAddress);
-
-  return basePrice;
-}
-
-export function getUnderlyingCoinFromPool(
-  poolAddress: Address,
-  curveRegistry: CurveRegistryContract
-): Address {
-  const coinsArray = curveRegistry.try_get_underlying_coins(poolAddress);
-
-  let coins: Address[];
-
-  if (coinsArray.reverted) {
-    return constants.NULL.TYPE_ADDRESS;
-  } else {
-    coins = coinsArray.value;
-  }
-
-  // Use first coin from pool and if that is empty (due to error) fall back to second coin
-  let preferredCoinAddress = coins[0];
-  if (preferredCoinAddress.equals(constants.NULL.TYPE_ADDRESS)) {
-    preferredCoinAddress = coins[1];
-  }
-
-  return preferredCoinAddress;
-}
-
-export function getVirtualPrice(
-  curveLpTokenAddress: Address,
-  curveRegistry: CurveRegistryContract
-): BigDecimal {
-  const virtualPrice = utils
-    .readValue<BigInt>(
-      curveRegistry.try_get_virtual_price_from_lp_token(curveLpTokenAddress),
-      constants.BIGINT_ZERO
-    )
-    .toBigDecimal();
-
-  return virtualPrice;
-}
-
-export function getPriceUsdcRecommended(
-  tokenAddress: Address
-): CustomPriceType {
-  return getUsdPricePerToken(tokenAddress);
-}
+import { getUsdPricePerToken } from "..";
+import * as utils from "../common/utils";
+import * as constants from "../common/constants";
+import { CustomPriceType } from "../common/types";
+import { BigInt, Address, BigDecimal } from "@graphprotocol/graph-ts";
+import { CurvePool as CurvePoolContract } from "../../../generated/templates/Pair/CurvePool";
+import { CurveRegistry as CurveRegistryContract } from "../../../generated/templates/Pair/CurveRegistry";
+
+export function isCurveLpToken(lpAddress: Address): bool {
+  const poolAddress = getPoolFromLpToken(lpAddress);
+  if (poolAddress.notEqual(constants.NULL.TYPE_ADDRESS)) return true;
+
+  return false;
+}
+
+export function getPoolFromLpToken(lpAddress: Address): Address {
+  let config = utils.getConfig();
```

Comment:
I'd say to correct the lint issues in this PR, seems like a lot of them are from added code not existing. So it would not make the PR messier